e2etest/pkg/config/validate.go

```diff
@@ -14,6 +14,23 @@ import (
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
+func ValidateAnnotateIPAllocatedFromPool(addressPools []metallbv1beta1.IPAddressPool, annotations map[string]string) error {
+	found := false
+	if poolName, ok := annotations["metallb.universe.tf/ip-allocated-from-pool"]; ok {
```

Comment:
I'd return error early and avoid the extra nesting
poolName, ok := ... 
if !ok {
    return error
}

At that point you can return instead of break-ing and return the error as the last return statement (and get rid of found)

Reply:
Thanks!